docs/design/event-streams.md

```diff
@@ -0,0 +1,246 @@
+# Kotlin Event Streaming Design
+
+* **Type**: Design
+* **Author(s)**: Aaron Todd
+
+# Abstract
+
+This document covers the client interfaces to be generated for requests/responses with event stream payloads 
+(`@streaming` Smithy trait applied to a `union` shape). See the [event stream spec](https://awslabs.github.io/smithy/1.0/spec/core/stream-traits.html#event-streams).
+
+Reference the additional documents listed in the Appendix for surrounding context on Smithy.
+
+
+# Design
+
+## Client Interface
+
+### Model
+
+We will use the examples from the event stream spec to explore the client interfaces to be generated:
+
+
+**Event stream on an input**
+
+```
+namespace smithy.example
+
+operation PublishMessages {
+    input: PublishMessagesInput
+}
+
+@input
+structure PublishMessagesInput {
+    room: String,
+    messages: PublishEvents,
+}
+
+@streaming
+union PublishEvents {
+    message: Message,
+    leave: LeaveEvent,
+}
+
+structure Message {
+    message: String,
+}
+
+structure LeaveEvent {}
+```
+
+**Event stream on an output**
+
+```
+namespace smithy.example
+
+operation SubscribeToMovements {
+    input: SubscribeToMovementsInput,
+    output: SubscribeToMovementsOutput
+}
+
+@input
+structure SubscribeToMovementsInput {}
+
+@output
+structure SubscribeToMovementsOutput {
+    movements: MovementEvents,
+}
+
+@streaming
+union MovementEvents {
+    up: Movement,
+    down: Movement,
+    left: Movement,
+    right: Movement,
+    throttlingError: ThrottlingError
+}
+
+structure Movement {
+    velocity: Float,
+}
+
+/// An example error emitted when the client is throttled
+/// and should terminate the event stream.
+@error("client")
+@retryable(throttling: true)
+structure ThrottlingError {}
+```
+
+
+### Event Stream Type Representation
+
+The members of an operation input or output that target a stream will be represented using by an asynchronous [Flow](https://kotlinlang.org/docs/reference/coroutines/flow.html) 
```

Comment:
correctness

in "The members of an operation input or output that target a stream will be represented using by an asynchronous [Flow]" seems like "by" is not needed.  "The members of an operation input or output that target a stream will be represented using an asynchronous [Flow]" or maybe "with".